File: 326. Power of Three.py
"""
class Solution:
    def isPowerOfThree(self, n: int) -> bool:
        if n <= 0:
            return False

        for i in range(0, 1000000):
            if 3**i == n:
                return True
            if 3**i > n:
                return False

        return False
"""


class Solution:
    def reculsive(self, n):
        n, rem = divmod(n, 3)
        
        if rem != 0:
            return False
        if n == 1:
            return True
        return self.reculsive(n)

    def isPowerOfThree(self, n: int) -> bool:
        if n <= 0:
            return False
        if n == 1:
            return True
        return self.reculsive(n)

File: 326. Power of Three.rs
impl Solution {
    pub fn reculsive(n: i32) -> bool {
        let quotient = n / 3;
        let remainder = n % 3;

        if remainder != 0 {
            return false;
        }

        if quotient == 1 {
            return true;
        }

        Self::reculsive(quotient)
    }

    pub fn is_power_of_three(n: i32) -> bool {
        if n <= 0 {
            return false;
        }
        if n == 1 {
            return true;
        }
        Self::reculsive(n)
    }
}
